### PR TITLE
Converts com_templates hardcoded BS modals to use JLayout/modal + Fix

### PR DESCRIPTION
**Pull request by cyrez**
Pull Request for Issue #9888 .
#### Summary of Changes
- addition of a clear div for modal-footer to fix display issue reported in #9888 and closed #9897
- refactory of the com_templates hardcoded bootsrap modals in default.php, to use JLayouts/modal (the issue was not using the Jlayout modal which includes core hack script to fix BS modal issues)
- addition of each modal body and footer in a separated template
- code style and better handler to detect when auto scrolling to be added and adjust modal height on small devices
#### Testing Instructions

**TEST on current STAGING ONLY!**
See #9888 (comment) for screenshots showing issue, and to help testing this patch.
In addition, test that all other modals (batch, users, versions...) are still working as before.
Same testing requirement as for PR https://github.com/joomla/joomla-cms/pull/9817#issue-147152727

**Note on com_templates admin**
In component templates, admin side, modal buttons to be tested:
- Copy template (collapseModal)
- Rename File (renameModal)
- Delete File (deleteModal)
- New File (fileModal)
- Manage Folders (folderModal)
- Resize Image (resizeModal)

A code review by PLT is welcome! (as addition of 12 news view files...)

### CODE REVIEW

**Comment by franz-wohlkoenig** [comment]
I have tested this item :white_check_mark: successfully on b041dfb2cb3b61acf47e71fe48fdffd04000e24b<br /><br />- Test like above described on com_template, Protostar (Hathor looks like don't use Modals).
- Test batch on users, Note Categories, Menu Items, Articles, -Categories, Banners, -Categories, Contacts,  -Categories, News Feeds, -Categories, Tags, Modules.
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9908">issues.joomla.org/joomla-cms/9908</a>.</sub>

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on b041dfb2cb3b61acf47e71fe48fdffd04000e24b<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9908">issues.joomla.org/joomla-cms/9908</a>.</sub>

**Comment by brianteeman** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9908">issues.joomla.org/joomla-cms/9908</a>.</sub>

File: layouts/joomla/modal/main.php [hunk 1]
```diff
@@ -74,20 +74,21 @@
 $script[] = "       $('body').addClass('modal-open');";
 
 // Get height of the modal elements.
-$script[] = "       var modalHeaderHeight = $('div.modal-header:visible').outerHeight(true);";
-$script[] = "       var modalBodyHeightOuter = $('div.modal-body:visible').outerHeight(true);";
-$script[] = "       var modalBodyHeight = $('div.modal-body:visible').height();";
-$script[] = "       var modalFooterHeight = $('div.modal-footer:visible').outerHeight(true);";
+$script[] = "       var modalHeight = $('div.modal:visible').outerHeight(true),";
+$script[] = "           modalHeaderHeight = $('div.modal-header:visible').outerHeight(true),";
```
Comment (rdeutz) [inline layouts/joomla/modal/main.php]: why did you remove all var declarations?

Comment (JoomliC) [inline layouts/joomla/modal/main.php]: To follow the Joomla coding standard for Multiple Variable Declarations: https://joomla.github.io/coding-standards/?coding-standards/chapters/javascript.md#syntax-indentation

But no var declarations removed ;-)

Comment (rdeutz) [inline layouts/joomla/modal/main.php]: oh missed the comma :-)
